frontend/src/modules/Traces/TraceGanttChart/TraceGanttChart.tsx

```diff
@@ -30,34 +42,42 @@ const TraceGanttChart = ({
 	selectedSpan,
 	resetZoom,
 	setSpanTagsInfo,
+	globalEndTime,
 }: TraceGanttChartProps) => {
 	let checkStrictly = true;
-	const [selectedRows, setSelectedRows] = useState([]);
-	const [clickedSpanData, setClickedSpanData] = useState(clickedSpan);
-	const [defaultExpandedRows, setDefaultExpandedRows] = useState([]);
+	const [selectedRows, setSelectedRows] = useState<Key[]>([]);
+	const [clickedSpanData, setClickedSpanData] = useState<pushDStree>(
+		clickedSpan,
+	);
+	const [defaultExpandedRows, setDefaultExpandedRows] = useState<Key[]>([]);
 	const [sortedTreeData, setSortedTreeData] = useState(treeData);
 	const [isReset, setIsReset] = useState(false);
 	const [tabsContainerWidth, setTabsContainerWidth] = useState(0);
-	const tableRef = useRef("");
-	let tabsContainer = document.querySelector("#collapsable .ant-tabs-nav-list");
+	let tabsContainer = document.querySelector<HTMLElement>(
```

Comment:
as these are expensive operation, why we need it?
is any alternative?

Reply:
Can't think of any alternative way to do this. Refs won't work here as it is an internal component of antd Table. Lmk if you any suggestions.